hooks/useProceed.ts

```diff
@@ -1,27 +1,45 @@
 'use client'
 
-import { getNextLessonPath } from 'lib/progress'
-import { useRouter } from 'next/navigation'
-import { usePathData, useLocalizedRoutes } from 'hooks'
-import { lessons } from 'content'
+import { useRouter, usePathname } from 'next/navigation'
+import { useLocalizedRoutes, usePathData } from 'hooks'
 import useEnvironment from './useEnvironment'
+import { useAtom, useAtomValue, useSetAtom } from 'jotai'
+import {
+  getLessonKey,
+  getNextLessonUsingChapterIdAndLessonName,
+  markLessonAsCompleteAtom,
+  nextLessonPathAtom,
+  progressToNextLessonAtom,
+  syncedCourseProgressAtom,
+} from 'state/progressState'
 
 export default function useProceed() {
+  const { chapterId, lessonId: lessonName } = usePathData()
   const router = useRouter()
-  const { chapterId, lessonId } = usePathData()
   const { isDevelopment } = useEnvironment()
   const routes = useLocalizedRoutes()
-
-  const chapterLessons = lessons?.[chapterId]
-
-  const lesson = chapterLessons?.[lessonId]?.metadata ?? null
-  const currentLessonKey = lesson?.key ?? 'CH1INT1'
+  const nextLessonPath = useAtomValue(nextLessonPathAtom)
+  const [courseProgress, setCourseProgress] = useAtom(syncedCourseProgressAtom)
+  const progressToNextLesson = useSetAtom(progressToNextLessonAtom)
   const queryParams = isDevelopment ? '?dev=true' : ''
+  const nextLessonUsingCurrentRoute = getNextLessonUsingChapterIdAndLessonName(
+    chapterId,
+    lessonName,
+    courseProgress
+  )
+  const currentLessonId = getLessonKey(chapterId, lessonName)
+  const markLessonAsComplete = useSetAtom(markLessonAsCompleteAtom)
 
-  const Proceed = async () => {
-    const nextLessonPath = getNextLessonPath(currentLessonKey)
-    const route = routes.chaptersUrl + nextLessonPath + queryParams
-
+  const Proceed = () => {
+    let route
+    if (isDevelopment && nextLessonUsingCurrentRoute) {
+      route =
+        routes.chaptersUrl + nextLessonUsingCurrentRoute?.path + queryParams
+      markLessonAsComplete(currentLessonId)
+    } else {
+      route = routes.chaptersUrl + nextLessonPath + queryParams
```

Comment:
Yeah, it seems like changing this to 
`route = routes.chaptersUrl + nextLessonUsingCurrentRoute?.path + queryParams`
fixes it but then the conditional doesn't seem to have a reason to exist